src/components/parameter/EditParameters.js

```diff
@@ -0,0 +1,241 @@
+
+    onChange(updatedParameters);
+  };
+
+  const addParam = () => {
+    const newParam = {
+      name: '',
+      type: PARAMETER_TYPES.INTEGER_INPUT,
+      value: PARAMETER_TYPES_DEFAULT[PARAMETER_TYPES.INTEGER_INPUT],
+    };
+
+    onChange([...parameters, newParam]);
+  };
+
+  const updateName = (name, paramIdx) => {
+    updateParam({ ...parameters[paramIdx], name }, paramIdx);
+  };
+
+  const updateType = (type, paramIdx) => {
+    updateParam(
+      {
+        ...parameters[paramIdx],
+        type,
+        value: PARAMETER_TYPES_DEFAULT[type],
+      },
+      paramIdx,
+    );
+  };
+
+  const updateValue = (value, paramIdx) => {
+    updateParam({ ...parameters[paramIdx], value }, paramIdx);
+  };
+
+  const renderParam = (param, paramIdx) => {
+    const { type, value } = param;
+    const invalid = !isParameterValid(param);
+    switch (type) {
+      case PARAMETER_TYPES.STRING_INPUT: {
+        return (
+          <TextField
+            value={value}
+            onChange={(event) => updateValue(event.target.value, paramIdx)}
+            label={t('Default value')}
+          />
+        );
+      }
+      case PARAMETER_TYPES.INTEGER_INPUT: {
+        return (
+          <TextField
+            value={value}
+            onChange={(event) => updateValue(event.target.value, paramIdx)}
+            error={invalid}
+            helperText={invalid && t('Please provide an integer')}
+            label={t('Default value')}
+          />
+        );
+      }
+      case PARAMETER_TYPES.FLOAT_INPUT: {
+        return (
+          <TextField
+            value={value}
+            onChange={(event) => updateValue(event.target.value, paramIdx)}
+            error={invalid}
+            helperText={invalid && t('Please provide a number')}
+            label={t('Default value')}
+          />
+        );
+      }
+      case PARAMETER_TYPES.FIELD_SELECTOR:
+        return (
+          <FieldSelectorButton
+            param={param}
+            onChange={(newValue) => updateValue(newValue, paramIdx)}
+            className={classes.parameterTopMargin}
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div id={id} className={className}>
+      <Typography variant="h6">{t('Parameters')}</Typography>
+      {(parameters.some(({ name }) => name.length === 0) && (
+        <Alert severity="error">{t("Parameter names can't be empty")}</Alert>
+      )) ||
+        (!areParametersNamesUnique(parameters) && (
+          <Alert severity="error">
+            {t("Parameters can't have the same name")}
+          </Alert>
+        ))}
+      <div className={classes.parametersContent}>
+        {parameters?.map((param, paramIdx) => {
+          const { name, type } = param;
+          const invalidName = name.length > 0 && !isParameterNameValid(param);
+          return (
+            // eslint-disable-next-line react/no-array-index-key
+            <Grid container spacing={1} key={paramIdx}>
```

Comment:
With the name as key, not only the console complains, but very annoying conflicts occur when trying to modify the name (like modifying another textfield instead of the current one, not being focused on the field anymore after entering one character).
I went with both the name and the idx

Reply:
Actually eslint still complains if I try to use the index in the key. I'll keep using only index then and keep the eslint-disable (unless you see another solution :eyes: )